### ISSUE TITLE
[SUPPORT] - SSH Key Signing - Hashicorp Vault

### ISSUE THREAD

**Issue by hbzhost** [paramiko/paramiko#2311]
### Are you using paramiko as a client or server?

Client

### What feature(s) aren't working right?

SSH, Keys/auth

### What version(s) of paramiko are you using?

3.3.1

### What version(s) of Python are you using?

3.11.4

### What operating system and version are you using?

Mac M1

### If you're connecting as a client, which SSH server are you connecting to?

OpenSSH

### What are you trying to do with paramiko?

I'm trying to ssh into my server in order to execute certain commands using generate private RSA key and signed ssh key cert via Hashicorp Vault. 

### How are you trying to do it, and what's happening instead?

1. I first generate my private key and get public key that needs to be signed by Hashicorp Vault. 

```
def generate_ssh_key_pair():
    private_key = rsa.generate_private_key(backend=default_backend(), public_exponent=65537, key_size=2048)
    public_key = private_key.public_key()
    private_key_pem = private_key.private_bytes(
        encoding=serialization.Encoding.PEM,
        format=serialization.PrivateFormat.TraditionalOpenSSL,
        encryption_algorithm=serialization.NoEncryption()
    ).decode('utf-8')

```
2. Then I send the key to be signed by Hashicorp Vault using public key from generated key above.
3. Key is signed by Vault without any issues using hvac library.
4. I write both private key and generated signed certificate from Hashicorp Vault locally to separate files. 
5. Then I try to establish ssh connection with Paramiko client using below code. 
```
        ssh_client = paramiko.SSHClient()
        ssh_client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
        # Connect to the SSH server
        test_connection = ssh_client.connect(ssh_host, ssh_port, ssh_username, key_filename=[private_key_path, public_key_cert_path])
```
6. I'm getting "Error: Authentication failed" and I know at least pkey parameter when using regular ssh private key without signed cert works just fine. However, for some reason adding certificate to the key_filename params doesn't pass it along with private key in order to auth. 

Basically I'm trying to use IdentityFile and CertificateFile if you were to use your local ssh client.

**Comment by jun66j5** [comment]
> ```
>     test_connection = ssh_client.connect(ssh_host, ssh_port, ssh_username, key_filename=[private_key_path, public_key_cert_path])
> ```

It is caused by the wrong given value for `key_filename` parameter.

> `key_filename` may contain OpenSSH public certificate paths as well as regular private-key paths; when files ending in `-cert.pub` are found, they are assumed to match a private key, and both components will be loaded. (The private key itself does not need to be listed in `key_filename` for this to occur - just the certificate.)

See document of the `key_parameter` at https://docs.paramiko.org/en/latest/api/client.html?highlight=key_parameter%20cert.pub#paramiko.client.SSHClient.connect

**Comment by jun66j5** [comment]
> ```
>     test_connection = ssh_client.connect(ssh_host, ssh_port, ssh_username, key_filename=[private_key_path, public_key_cert_path])
> ```

It is caused by the wrong given value for `key_filename` parameter and file name of signed certificate.

> `key_filename` may contain OpenSSH public certificate paths as well as regular private-key paths; when files ending in `-cert.pub` are found, they are assumed to match a private key, and both components will be loaded. (The private key itself does not need to be listed in `key_filename` for this to occur - just the certificate.)

See document of the `key_parameter` at https://docs.paramiko.org/en/latest/api/client.html?highlight=key_parameter%20cert.pub#paramiko.client.SSHClient.connect
